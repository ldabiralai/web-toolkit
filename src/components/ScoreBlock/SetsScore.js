import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import * as colors from '../../colors';
import { fontFamilies } from '../../typography';
import * as breakpoints from '../../breakpoints';

const StyledTeamWrapper = styled.div`
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const StyledTeamName = styled.div`
  color: ${props => (props.hasWon === false ? colors.coreNeutral4 : colors.coreLightMinus1)};
  font-family: ${fontFamilies.alphaHeadline};
  font-size: 1em;
  flex-basis: 60%;
  text-transform: uppercase;
  span {
    font-family: ${fontFamilies.interUi};
    font-size: 1em;
  }
`;

export const StyledTeamSet = styled.span`
  color: ${props => (props.won ? colors.coreLightMinus1 : colors.coreNeutral4)};
  font-family: ${fontFamilies.alphaHeadline};
  font-weight: bold;
  font-size: 1em;
  margin: 0 0.4em;
  &:last-child {
    margin-right: 0px;
  }
  sup {
    font-size: 0.55em;
    vertical-align: super;
    position: relative;
    top: -0.4em;
    right: -0.2em;
  }
  ${breakpoints.small(css`
    margin: 0 0.8em;
  `)};
`;

const StyledTeamSets = styled.div`
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
`;

const Team = ({ teamData }) => {
  const { playerOneName, playerTwoName, sets, isServing, hasWon } = teamData;
  return (
    <StyledTeamWrapper>
      <StyledTeamName hasWon={hasWon}>
        <div>
          {playerOneName}
          {hasWon && <span> ✓</span>}
          {isServing && <span> •</span>}
        </div>
        <div>{playerTwoName}</div>
      </StyledTeamName>
      <StyledTeamSets>
        {sets &&
          sets.map(({ set, score, tie, ...props }) => (
            <StyledTeamSet key={`set-${set}`} {...props}>
              {score}
              {tie && <sup>{tie}</sup>}
            </StyledTeamSet>
          ))}
      </StyledTeamSets>
    </StyledTeamWrapper>
  );
};

const StyledWrapper = styled.div`
  width: 100%;
  ${props =>
    props.baseFontSize
      ? css`
          font-size: ${props.baseFontSize};
        `
      : css`
    font-size: 8px
    ${breakpoints.small(css`
      font-size: 14px;
    `)};
    ${breakpoints.medium(css`
      font-size: 24px;
    `)};
    ${breakpoints.large(css`
      font-size: 32px;
    `)};
    };
  `}
`;

export const StyledSpacer = styled.hr`
  display: block;
  height: 1px;
  border: 0;
  border-top: 1px solid;
  padding: 0;
  border-top-color: ${colors.coreLightMinus1};
`;

const SetsScore = ({ data, baseFontSize }) => (
  <StyledWrapper baseFontSize={baseFontSize} data-test="sets-score-wrapper">
    <Team teamData={data.topTeam} />
    <StyledSpacer />
    <Team teamData={data.bottomTeam} />
  </StyledWrapper>
);

const teamDataType = PropTypes.shape({
  hasWon: PropTypes.bool,
  isServing: PropTypes.bool,
  playerOneName: PropTypes.string.isRequired,
  playerTwoName: PropTypes.string,
  sets: PropTypes.arrayOf(
    PropTypes.shape({
      set: PropTypes.number,
      score: PropTypes.number,
      tie: PropTypes.number,
      won: PropTypes.bool,
    })
  ),
});

Team.propTypes = {
  teamData: teamDataType.isRequired,
};

SetsScore.defaultProps = {
  baseFontSize: null,
};

SetsScore.propTypes = {
  data: PropTypes.shape({
    bottomTeam: teamDataType,
    topTeam: teamDataType,
  }).isRequired,
  baseFontSize: PropTypes.string,
};

export default SetsScore;
