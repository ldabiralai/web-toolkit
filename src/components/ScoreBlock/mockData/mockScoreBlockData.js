export const pastMatchData = {
  topTeam: {
    playerOneName: 'Federer',
    sets: [
      { set: 1, score: 6, won: true, tie: null },
      { set: 2, score: 6, won: true, tie: null },
      { set: 3, score: 6, won: false, tie: 2 },
      { set: 4, score: 6, won: true, tie: null },
      { set: 5, score: 3, won: false, tie: null },
    ],
    hasWon: true,
    isServing: false,
  },
  bottomTeam: {
    playerOneName: 'Nadal',
    sets: [
      { set: 1, score: 3, won: false, tie: null },
      { set: 2, score: 3, won: true, tie: null },
      { set: 3, score: 7, won: true, tie: 7 },
      { set: 4, score: 3, won: false, tie: null },
      { set: 5, score: 6, won: true, tie: null },
    ],
    hasWon: false,
    isServing: false,
  },
};

export const liveMatchData = {
  topTeam: {
    playerOneName: 'MENENDEZ-MACEIRAS',
    playerTwoName: 'Rahman MAKSSOUD',
    sets: [
      { set: 1, score: 6, won: true, tie: null },
      { set: 2, score: 6, won: true, tie: null },
      { set: 3, score: 6, won: false, tie: 2 },
      { set: 4, score: 6, won: true, tie: null },
      { set: 5, score: 3, won: false, tie: null },
    ],
    hasWon: null,
    isServing: false,
  },
  bottomTeam: {
    playerOneName: 'MENENDEZ-MACEIRAS',
    playerTwoName: 'DAVIDOVICH FOKINA',
    sets: [
      { set: 1, score: 3, won: false, tie: null },
      { set: 2, score: 3, won: true, tie: null },
      { set: 3, score: 7, won: true, tie: 7 },
      { set: 4, score: 3, won: false, tie: null },
      { set: 5, score: 6, won: true, tie: null },
    ],
    hasWon: null,
    isServing: true,
  },
};

export const liveMatchDataSet = {
  topTeam: {
    playerOneName: 'A. MENENDEZ-MACEIRAS',
    playerTwoName: 'A. Rahman MAKSSOUD',
    sets: [
      { set: 1, score: 6, won: true, tie: null },
      { set: 2, score: 6, won: true, tie: null },
      { set: 3, score: 6, won: false, tie: 2 },
      { set: 4, score: 6, won: true, tie: null },
      { set: 5, score: 3, won: false, tie: null },
    ],
    hasWon: null,
    isServing: false,
  },
  bottomTeam: {
    playerOneName: 'A. MENENDEZ-MACEIRAS',
    playerTwoName: 'A. DAVIDOVICH FOKINA',
    sets: [
      { set: 1, score: 3, won: false, tie: null },
      { set: 2, score: 3, won: true, tie: null },
      { set: 3, score: 7, won: true, tie: 7 },
      { set: 4, score: 3, won: false, tie: null },
      { set: 5, score: 6, won: true, tie: null },
    ],
    hasWon: null,
    isServing: true,
  },
};
