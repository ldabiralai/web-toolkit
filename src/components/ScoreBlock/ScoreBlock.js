import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import * as colors from '../../colors';
import * as breakpoints from '../../breakpoints';
import SetsScore, { StyledSpacer } from './SetsScore';
import { fontFamilies } from '../../typography';
import circleWithCross from '../../assets/circle-with-cross.svg';
import greenCircle from '../../assets/green-circle.svg';
import { Link } from '../..';

const StyledClickableWrapper = styled(Link)`
  width: 100%;
  display: flex;
  align-items: stretch;
  text-decoration: none;
  font-size: 14px;
  color: inherit;
`;

export const StyledButton = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${props => colors[props.color]};
  font-family: ${fontFamilies.interUi};
  text-align: center;
  text-transform: uppercase;
  color: ${colors.coreLightMinus1};
  letter-spacing: 1px;
  line-height: 16px;
  width: ${props => props.width};
  min-width: 58px;
  ${breakpoints.small(css`
    padding: 0 4px;
  `)};
  ${props =>
    css`
      &:after {
        content: '\u2192';
      }
      ${breakpoints.small(css`
        &:after {
          content: ${props.contentText};
        }
      `)};
    `}
`;

const StyledScoreWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  padding-right: 16px;
  ${props =>
    !props.hasLeftCircle &&
    css`
      padding-left: 16px;
    `}
  padding-top: 16px;
  padding-bottom: 16px;
  flex-basis: 400px;
  flex-grow: 1;
  background-color: rgba(79, 82, 106, 0.5);
  ${StyledSpacer} {
    border-top-color: ${colors.midnightExpress2};
  }
`;

const StyledPlayButton = styled(StyledButton)`
  font-size: 24px;
  display: none;
  ${breakpoints.small(css`
    display: inherit;
  `)};
  &:after {
    content: none;
  }
`;

const StyledBigDot = styled.div`
  display: flex;
  align-items: center;
  font-family: ${fontFamilies.interUi};
  color: ${colors.featureThreeBase};
  margin: 0 11px;
  font-size: 72px;
  ${breakpoints.medium(css`
    margin: 0 29px;
  `)};
  ${breakpoints.large(css`
    font-size: 78px;
  `)};
`;

const ScoreBlock = ({ matchUrl, data, isLive, isWatchable, displayLeftCircle }) => {
  const hasLeftCircle = ['won', 'lost'].includes(displayLeftCircle);
  return (
    <StyledClickableWrapper href={matchUrl} data-test="clickable-scoreblock-wrapper">
      <StyledScoreWrapper hasLeftCircle={hasLeftCircle}>
        {hasLeftCircle && (
          <StyledBigDot>
            {displayLeftCircle === 'won' ? (
              <img src={greenCircle} alt="won match icon" />
            ) : (
              <img src={circleWithCross} alt="lost match icon" />
            )}
          </StyledBigDot>
        )}
        <SetsScore data={data} baseFontSize="14px" />
      </StyledScoreWrapper>
      {isLive ? (
        <>
          {isWatchable && (
            <StyledPlayButton color="actionTwoLightPlus1" width="68px">
              &#9658;
            </StyledPlayButton>
          )}
          <StyledButton color="venetianRed" contentText="'LIVE'" width="68px" />
        </>
      ) : (
        <StyledButton color="actionOneDarkBase" contentText="'MATCH INFO'" width="144px" />
      )}
    </StyledClickableWrapper>
  );
};

const teamDataType = PropTypes.shape({
  hasWon: PropTypes.bool,
  isServing: PropTypes.bool,
  playerOneName: PropTypes.string,
  playerTwoName: PropTypes.string,
  sets: PropTypes.arrayOf(
    PropTypes.shape({
      set: PropTypes.number,
      score: PropTypes.number,
      tie: PropTypes.number,
      won: PropTypes.bool,
    })
  ),
});

ScoreBlock.defaultProps = {
  displayLeftCircle: false,
};

ScoreBlock.propTypes = {
  matchUrl: PropTypes.string.isRequired,
  data: PropTypes.shape({
    bottomTeam: teamDataType,
    topTeam: teamDataType,
  }).isRequired,
  isLive: PropTypes.bool.isRequired,
  isWatchable: PropTypes.bool.isRequired,
  displayLeftCircle: PropTypes.oneOf(['won', 'lost', false]),
};

export default ScoreBlock;
