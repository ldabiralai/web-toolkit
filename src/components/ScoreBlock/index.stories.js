import React from 'react';
import styled from 'react-emotion';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { object, boolean, select } from '@storybook/addon-knobs';
import { ScoreBlocks } from '../..';
import { pastMatchData, liveMatchData, liveMatchDataSet } from './mockData/mockScoreBlockData';

const MATCH_URL =
  'https://www.eurosport.fr/tennis/barcelone-1/2019/live-rafael-nadal-leonardo-mayer_mtc1105580/live.shtml';

const Wrapper = styled.div`
  max-width: 633px;
`;

const scoreBlockStories = storiesOf('Components|Score Block', module);

scoreBlockStories.add(
  'SetsScore',
  withInfo()(() => (
    <Wrapper css={{ maxWidth: '1067px' }}>
      <ScoreBlocks.SetsScore data={object('score data', liveMatchDataSet)} />
    </Wrapper>
  ))
);

scoreBlockStories.add(
  `ScoreBlock - info`,
  withInfo()(() => (
    <Wrapper>
      <ScoreBlocks.ScoreBlock
        matchUrl={MATCH_URL}
        data={object('score data', pastMatchData)}
        isLive={boolean('isLive', false)}
        isWatchable={boolean('isWatchable', false)}
        displayLeftCircle={select('displayLeftCircle', ['won', 'lost', false], false)}
      />
    </Wrapper>
  ))
);
scoreBlockStories.add(
  `ScoreBlock - live`,
  withInfo()(() => (
    <Wrapper>
      <ScoreBlocks.ScoreBlock
        matchUrl={MATCH_URL}
        data={object('score data', liveMatchData)}
        isLive={boolean('isLive', true)}
        isWatchable={boolean('isWatchable', true)}
        displayLeftCircle={select('displayLeftCircle', ['won', 'lost', false], false)}
      />
    </Wrapper>
  ))
);
