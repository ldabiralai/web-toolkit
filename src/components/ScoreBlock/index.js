import SetsScore from './SetsScore';
import ScoreBlock from './ScoreBlock';

export default { SetsScore, ScoreBlock };
