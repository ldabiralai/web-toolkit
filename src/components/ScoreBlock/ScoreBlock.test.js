import React from 'react';
import { shallow, mount } from 'enzyme';
import ScoreBlock, { StyledButton } from './ScoreBlock';
import { liveMatchData, pastMatchData } from './mockData/mockScoreBlockData';
import greenCircleSVG from '../../assets/green-circle.svg';
import crossSVG from '../../assets/circle-with-cross.svg';

describe('<ScoreBlock />', () => {
  it('Renders a ScoreBlock component with expected snapshot', () => {
    const wrapper = shallow(<ScoreBlock data={pastMatchData} matchUrl="www.eurosport.fr" isWatchable isLive />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should display '->' if isLive", () => {
    const wrapper = mount(<ScoreBlock isLive data={liveMatchData} matchUrl="www.google.fr" isWatchable={false} />);
    const styledButton = wrapper.find(StyledButton);
    expect(styledButton).toHaveStyleRule('content', "'\u2192'");
  });

  it('should display a green circle if displayLeftCircle is set to won', () => {
    const wrapper = mount(
      <ScoreBlock isLive data={liveMatchData} matchUrl="www.google.fr" isWatchable={false} displayLeftCircle="won" />
    );
    const greenCircle = wrapper.find('img');
    expect(greenCircle.prop('src')).toEqual(greenCircleSVG);
  });

  it('should display a grey circle with cross if displayLeftCircle is set to lost', () => {
    const wrapper = mount(
      <ScoreBlock isLive data={liveMatchData} matchUrl="www.google.fr" isWatchable={false} displayLeftCircle="lost" />
    );
    const circle = wrapper.find('img');
    expect(circle.prop('src')).toEqual(crossSVG);
  });
});
