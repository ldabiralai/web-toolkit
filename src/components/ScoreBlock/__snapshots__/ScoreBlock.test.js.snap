// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreBlock /> Renders a ScoreBlock component with expected snapshot 1`] = `
<Styled(Link)
  data-test="clickable-scoreblock-wrapper"
  href="www.eurosport.fr"
  linkComponent={[Function]}
>
  <Styled(div)
    hasLeftCircle={false}
  >
    <SetsScore
      baseFontSize="14px"
      data={
        Object {
          "bottomTeam": Object {
            "hasWon": false,
            "isServing": false,
            "playerOneName": "Nadal",
            "sets": Array [
              Object {
                "score": 3,
                "set": 1,
                "tie": null,
                "won": false,
              },
              Object {
                "score": 3,
                "set": 2,
                "tie": null,
                "won": true,
              },
              Object {
                "score": 7,
                "set": 3,
                "tie": 7,
                "won": true,
              },
              Object {
                "score": 3,
                "set": 4,
                "tie": null,
                "won": false,
              },
              Object {
                "score": 6,
                "set": 5,
                "tie": null,
                "won": true,
              },
            ],
          },
          "topTeam": Object {
            "hasWon": true,
            "isServing": false,
            "playerOneName": "Federer",
            "sets": Array [
              Object {
                "score": 6,
                "set": 1,
                "tie": null,
                "won": true,
              },
              Object {
                "score": 6,
                "set": 2,
                "tie": null,
                "won": true,
              },
              Object {
                "score": 6,
                "set": 3,
                "tie": 2,
                "won": false,
              },
              Object {
                "score": 6,
                "set": 4,
                "tie": null,
                "won": true,
              },
              Object {
                "score": 3,
                "set": 5,
                "tie": null,
                "won": false,
              },
            ],
          },
        }
      }
    />
  </Styled(div)>
  <Styled(div)
    color="actionTwoLightPlus1"
    width="68px"
  >
    â–º
  </Styled(div)>
  <Styled(div)
    color="venetianRed"
    contentText="'LIVE'"
    width="68px"
  />
</Styled(Link)>
`;
